File: 0058-length-of-last-word/0058-length-of-last-word.php
class Solution {

    /**
     * @param String $s
     * @return Integer
     */
    function lengthOfLastWord($s) {
        $sArr = str_split($s);
        $ans = 0;
            
        for($i = count($sArr) - 1; $i >= 0; $i--){
            if($sArr[$i] == ' '){
                if($ans){
                    break;
                }
                continue;
            }
            
            $ans++;
        }
        
        return $ans;
    }
}

File: 0058-length-of-last-word/0058-length-of-last-word.go
func lengthOfLastWord(s string) int {
    len := 0
    reset := false
    
    for _, val := range s {
        if val == ' ' {
            reset = true
            continue
        }
        
        if reset == true {
            len = 0
            reset = false
        }
        
        len++
    }
    
    return len
}